evdev.lisp

```diff
@@ -154,25 +154,47 @@ used outside of this package.")
       (116 . (:name power            :glyph nil))
       (117 . (:name kpequal          :glyph #\=))
       (118 . (:name kpplusminus      :glyph nil))
-      (119 . (:name pause            :glyph nil)))
-  :test #'equal
+      (119 . (:name pause            :glyph nil))
+      (277 . (:name btnforward      :glyph nil))
+      (278 . (:name btnback         :glyph nil)))      
+   :test #'equal
   :documentation "List of key code to key symbol name and printable character.
 Used to decode the code field of the Linux input_event struct defined in
 linux/include/uapi/linux/input.h.")
 
-(define-unsigned unsigned-long-int 8)
+(define-constant +input-abs-codes+
+ '((0 . (:name ABS_X))
+   (1 . (:name ABS_Y))
+   (24 . (:name ABS_PRESSURE))
+   (40 . (:name ABS_MISC)))
+  :test #'equal
+  :documentation "Absolute device values for pointer and tablet hardware."
+)
+
+(define-constant +input-rel-codes+
+ '((8 . (:name REL_WHEEL)))
+  :test #'equal
+  :documentation "Relative motion types."
+)
+
+
+(cond ((equal (machine-type) "X86") 
+       (define-unsigned unsigned-long-int 4))
+      ((equal (machine-type) "X86-64") 
+       (define-unsigned unsigned-long-int 8))
+      (t 4))
 (define-unsigned unsigned-short 2)
 (define-unsigned unsigned-int 4)
 
 (define-binary-class input-event-struct ()
-  ((tv_sec  :binary-type unsigned-long-int)
+  ((tv_sec  :binary-type unsigned-long-int)			     
    (tv_usec :binary-type unsigned-long-int)
    (type    :binary-type unsigned-short)
-   (code    :binary-type unsigned-short)
-   (value   :binary-type unsigned-int))
+   (code    :binary-type unsigned-short) 
+   (value   :binary-type unsigned-int)) 
   (:documentation "This is a verbaticm copy of the input_event struct defined in
 linux/include/uapi/linux/input.h. This is used to read in each event straight
-from raw evdev data."))
+from raw evdev data. Unix time values bit widths differ on 32/64bit systems."))
```

Comment:
We'll want to note here that the difference is based upon the return value of machine-type.